src/app/dataset-block/metadata-block/components/event-details/builder.events.ts

```diff
@@ -40,15 +49,120 @@ type SetPollingSourceScalarSections =
     | SetPollingSourceSection.PREPROCESS
     | SetPollingSourceSection.READ;
 
-abstract class EventSectionBuilder<TEvent> {
-    public abstract buildEventSections(event: TEvent): EventSection[];
+enum SetTransformSection {
+    INPUTS = "inputs",
+    TRANSFORM = "transform",
 }
 
-class SetPollingSourceSectionBuilder extends EventSectionBuilder<SetPollingSource> {
+type SetTransformScalarSections =
+    | SetTransformSection.INPUTS
+    | SetTransformSection.TRANSFORM;
+
+export const descriptorMapper = {
```

Comment:
I guess thie type of this is `Record<EventTypesCalar, Record<string, RowDescriptor>>

Reply:
Added type for this object